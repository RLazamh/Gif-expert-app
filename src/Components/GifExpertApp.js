import React, { useState } from 'react'
// import PropTypes from 'prop-types'
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const  GifExpertApp = ({defaultCategories = [] }) => {

    // const categories = ['anime','terror','heroes'];
    // if(defaultCategories.length === 0) defaultCategories = ['One Punch Man']
    const [categories, setCategories] =useState(defaultCategories);
    // --MI solucion 
    // const handleAdd = ()=>{
    //     setCategories(categories.concat('comedia'));
    // }
    // Solucion del profe
    // const handleAdd = ()=>{
    //     setCategories(['comedia',...categories]);
    // }
    // const handleAdd = () =>{
    //     setCategories((category)=>{
    //         return ['comedia',...category]
    //     })
    // }
    // const [state, setstate] = useState(initialState)
    return (
        <>
          <h2>GifExpertApp</h2>
          <AddCategory setCategories = {setCategories}/>
          <hr />

          {/* <button onClick={handleAdd}>Add</button>   */}
          <ol>
              {
                  categories.map((category)=>(
                        <GifGrid 
                        key = {category}
                        category = {category}/>
                    )
                  )
              }
          </ol>
        </>
    )
}

// GifExpertApp.propTypes = {

// }

export default GifExpertApp

