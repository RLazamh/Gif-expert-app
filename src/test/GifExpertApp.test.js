import { shallow } from "enzyme"
import GifExpertApp from "../Components/GifExpertApp"

describe('Test in the component <GifExpertApp />', () => {
    
    const wrapper = shallow(<GifExpertApp />)
    
    test('Should match with snapshot', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should show an array of categories', () => {
        const categories = ['One Punch', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    })
    

})
