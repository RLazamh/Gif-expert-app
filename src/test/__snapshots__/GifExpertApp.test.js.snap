// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the component <GifExpertApp /> Should match with snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in the component <GifExpertApp /> should show an array of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
